src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitForeachTests.cs

```diff
@@ -4796,6 +4885,51 @@ public async Task<int> DisposeAsync()
             CompileAndVerify(comp, expectedOutput: "MoveNextAsync DisposeAsync Done");
         }
 
+        [Fact]
+        public void PatternBasedDisposal_WithOptionalParameter()
+        {
+            string source = @"
+using System.Threading.Tasks;
+class C
+{
+    public static async Task Main()
+    {
+        await foreach (var i in new C())
+        {
+        }
+        System.Console.Write(""Done"");
+    }
+    public Enumerator GetAsyncEnumerator()
+    {
+        return new Enumerator();
+    }
+    public sealed class Enumerator
+    {
+        public async Task<bool> MoveNextAsync()
```

Comment:
>MoveNextAsync [](start = 32, length = 13)

Can this method be an extension method? Do we have tests covering this scenario?

Reply:
No, this cannot be an extension.

---
In reply to: [550214694](https://github.com/dotnet/roslyn/pull/50105#discussion_r550214694) [](ancestors = 550214694)